pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/ImmutableDictionaryReader.java

```diff
@@ -244,6 +252,6 @@ protected String getPaddedString(int dictId, byte[] buffer) {
   }
 
   protected byte[] getBuffer() {
-    return new byte[_numBytesPerValue];
+    return (_valueReader instanceof VarLengthBytesValueReaderWriter) ? null : new byte[_numBytesPerValue];
```

Comment:
Instance of can be avoided by using ThreadLocal byte[] that is initialized to null in the constructor (already has the `if/else`)?

Reply:
Fixed.